### PR TITLE
Add ability to expose derived state apis

### LINKED ISSUE #38: Support exposing derived state from an app that partner apps can consume.
The beauty of this library is that you can combine with multiple redux states and they can be able to lazy loaded and registered when appropriate. By being able to subscribe to a partner state, we get the ability to get the latest state from a partner app but the problem is that Redux ideally saved the smallest amount of data in the store, and use libraries like [reselect](https://github.com/reduxjs/reselect) can do computations and compute derived state. Making getting derived state from partner apps not possible.

If the GlobalStore was able to expose derived state APIs then the partner apps could listen to state changes, but also get derived state without knowing the structure of the partner state. Otherwise, listening to partner state changes would require another partner app to grab the state and massage the data as needed - while this _does_ work, it forces apps to be coupled and require to understand the structure of each others state, which is not desired.

I've tried chunking the selectors from partner apps, and lazy loading; however, `reselect` isn't meant to lazy load selectors, so it is rather difficult. Furthermore, it makes it hard to truly decouple apps.

I have submitted a PR to implement this feature here: https://github.com/microsoft/redux-micro-frontend/pull/39

Thank you so much for such a wonderful library! :)

### PR DESCRIPTION
**Pull request by datajoy-eric**
Resolves #38 

Open to any discussion in terms of naming, missing tests, missing functionality :)

Thank you!

### CODE REVIEW

File: src/global.store.ts [hunk 1]
```diff
@@ -334,6 +336,48 @@ export class GlobalStore implements IGlobalStore {
...
+     * Summary: Expose an API collection from a Partner-level that other partners can later consume. This allows partners to derive data without forcing partners to know the state structure.
+     * 
+     * @access public
+     * 
+     * @param {string} source Name of the application exposing an derived state API
+     * @param {Record<string, any>} api The collection of APIs of derived state selectors.
+     * @param {boolean} mergeApi If the source application already exposed an API set, merge the new API being passed in.
+     * 
+     */
+    ExposeDerivedState(source: string, api: Record<string, any>, mergeApi = false) {
```
Comment (patrickCode) [inline src/global.store.ts:349]: Rename suggestions
`ExposeDerivedState` --> `AddSelector`
`api` -->`selector`
`mergeApi` -> `mergeSelector`

File: src/global.store.ts [hunk 2]
```diff
@@ -334,6 +336,48 @@ export class GlobalStore implements IGlobalStore {
...
+     * Summary: Select derived state from a partner app using the API name
+     * 
+     * @access public
+     * 
+     * @param {string} partner Name of the partner application to select derived data from
+     * @param {string} interestedDerivedState The name of the API to select
+     * @param {any} defaultReturn If the partner app does not have that API exposed, return this default value instead of undefined.
+     * 
+     */
+    SelectPartnerDerivedState(partner: string, interestedDerivedState: string, defaultReturn?: any) {
```
Comment (patrickCode) [inline src/global.store.ts:369]: Rename suggestion
`SelectPartnerDeriviedState` -- `SelectPartnerState` or `SelectPartialState`
`interestedDerivedState` - `selected`
`defaultReturn` -- `default`

File: src/global.store.ts [hunk 3]
```diff
@@ -30,6 +31,7 @@ export class GlobalStore implements IGlobalStore {
         this._eagerPartnerStoreSubscribers = {};
         this._eagerUnsubscribers = {};
         this._actionLogger = new ActionLogger(_logger);
+        this._exposedDerivedState = {};
```
Comment (patrickCode) [inline src/global.store.ts:34]: Rename suggestion
`_exposedDerivedState` --> `selectors`

**Comment by patrickCode** [comment]
I have added a few renaming suggestions, please verify

File: src/global.store.ts (hunk 1 above)
Comment (datajoy-eric) [reply src/global.store.ts:349]: 100% these make far more sense as names. I opted to keep them as plural i.e. (`AddSelectors`, `selectors`, `mergeSelectors`) let me know your thoughts :)

File: src/global.store.ts (hunk 3 above)
Comment (datajoy-eric) [reply src/global.store.ts:34]: Renamed, i added the `_` at the front to keep it consistent with the rest of the properties in this class, let me know what you think :)

**Comment by datajoy-eric** [comment]
> I have added a few renaming suggestions, please verify

Thanks so much for the review, made the changes + left a few comments as well. Let me know what you think, I'd be happy to rename/change anything :)

**Comment by patrickCode** [comment]
> > I have added a few renaming suggestions, please verify
> 
> Thanks so much for the review, made the changes + left a few comments as well. Let me know what you think, I'd be happy to rename/change anything :)

I can't see the new changes, have you pushed the commits?

**Review by patrickCode — APPROVED** [review]
LGTM

**Pull request merged**